hazelcast/src/main/java/com/hazelcast/instance/impl/DefaultNodeExtension.java

```diff
@@ -237,6 +238,15 @@ protected void checkDynamicConfigurationPersistenceAllowed() {
         }
     }
 
+    protected void checkSqlCatalogPersistenceAllowed() {
+        Config config = node.getConfig();
+        if (config.getSqlConfig().isCatalogPersistenceEnabled()) {
+            if (!BuildInfoProvider.getBuildInfo().isEnterprise()) {
+                throw new IllegalStateException("SQL Catalog Persistence requires Hazelcast Enterprise Edition");
+            }
+        }
+    }
```

Comment:
Can we add a test which tests this path? I mean a test where a user starts the cluster without a license and then catches this exception.